pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -29,26 +29,21 @@ import (
 var _ entity.EntityStoreServer = &sqlEntityServer{}
 
 func ProvideSQLEntityServer(db db.EntityDBInterface /*, cfg *setting.Cfg */) (entity.EntityStoreServer, error) {
-	snode, err := snowflake.NewNode(rand.Int63n(1024))
-	if err != nil {
-		return nil, err
-	}
-
 	entityServer := &sqlEntityServer{
-		db:        db,
-		log:       log.New("sql-entity-server"),
-		snowflake: snode,
+		db:  db,
+		log: log.New("sql-entity-server"),
 	}
 
 	return entityServer, nil
 }
 
 type sqlEntityServer struct {
-	log       log.Logger
-	db        db.EntityDBInterface // needed to keep xorm engine in scope
-	sess      *session.SessionDB
-	dialect   migrator.Dialect
-	snowflake *snowflake.Node
+	log         log.Logger
+	db          db.EntityDBInterface // needed to keep xorm engine in scope
+	sess        *session.SessionDB
+	dialect     migrator.Dialect
+	snowflake   *snowflake.Node
+	broadcaster *Broadcaster[*entity.Entity]
```

Comment:
NIT – passing around pointers rather than values can cause very difficult-to-track bugs stemming from sharing memory. I recognize that passing values will create additional allocations but to me this looks like a premature optimization.

Reply:
this actually isn't easy to change because Entity is a protobuf object that includes a mutex, so I've stuck with pointers.